docs: adds correct updated documentation links

Zeegaths opened the pull request:
Added correct links to the docs in place of the missing docs' paths.
Fixes #32565

mabu44 reviewed (commented):
This makes it necessary to manually update the links with every release, or at least with every new version that contains changes to those documentation files.

Zeegaths commented:
> This makes it necessary to manually update the links with every release, or at least with every new version that contains changes to those documentation files.

it does, still trying to get a workaround that automatically updates it

Zeegaths commented:
> This makes it necessary to manually update the links with every release, or at least with every new version that contains changes to those documentation files.

i have update it so it fetches the docs based on build version

brunoerg commented on src/wallet/init.cpp:
```diff
@@ -66,7 +67,7 @@ void WalletInit::AddWalletOptions(ArgsManager& argsman) const
                                                             CURRENCY_UNIT, FormatMoney(CFeeRate{DEFAULT_PAY_TX_FEE}.GetFeePerK())), ArgsManager::ALLOW_ANY, OptionsCategory::WALLET);
 #ifdef ENABLE_EXTERNAL_SIGNER
-    argsman.AddArg("-signer=<cmd>", "External signing tool, see doc/external-signer.md", ArgsManager::ALLOW_ANY, OptionsCategory::WALLET);
+    argsman.AddArg("-signer=<cmd>", strprintf("External signing tool, see https://github.com/bitcoin/bitcoin/blob/v%d.%d/doc/external-signer.md", CLIENT_VERSION_MAJOR, CLIENT_VERSION_MINOR), ArgsManager::ALLOW_ANY, OptionsCategory::WALLET);
```
Running on master branch, I get: 
```sh
 -signer=<cmd>
       External signing tool, see
       https://github.com/bitcoin/bitcoin/blob/v29.99/doc/external-signer.md
```
...which is incorrect since there is no v29.99.

Zeegaths replied:
this could be because you are on developement version. Let me try set it to the latest stable version

brunoerg commented:
You can squash your commits according to https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#squashing-commits.

Zeegaths commented:
> You can squash your commits according to https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#squashing-commits.

thankyou for this

musaHaruna reviewed (commented):
I checked out v29.0 with the changes, I got: 
`  -cjdnsreachable
       If set, then this host is configured for CJDNS (connecting to fc00::/8
       addresses would lead us to the CJDNS network, see
       https://github.com/bitcoin/bitcoin/blob/v29.0/doc/cjdns.md)
       (default: 0)
`
which links correctly, the function `GetDocumentationUrl` looks good to me

Zeegaths commented:
Thankyou for the feedback

On Mon, 16 Jun 2025 at 15:09, musaHaruna ***@***.***> wrote:

> ***@***.**** commented on this pull request.
>
> I checked out v29.0 with the changes, I got:
> -cjdnsreachable If set, then this host is configured for CJDNS (connecting
> to fc00::/8 addresses would lead us to the CJDNS network, see
> https://github.com/bitcoin/bitcoin/blob/v29.0/doc/cjdns.md) (default: 0)
> which links correctly, the function GetDocumentationUrl looks good to me
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/bitcoin/bitcoin/pull/32699#pullrequestreview-2931880588>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AZVGWLGZSNRGKGVIZRYRJGL3D2XYVAVCNFSM6AAAAAB6ZXKXIOVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDSMZRHA4DANJYHA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

Zeegaths commented:
Thank you, I will do just that

On Mon, 16 Jun 2025 at 18:12, Winnie Gitau ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In src/init.cpp
> <https://github.com/bitcoin/bitcoin/pull/32699#discussion_r2150261914>:
>
> > @@ -157,7 +157,7 @@ static constexpr bool DEFAULT_STOPAFTERBLOCKIMPORT{false};
>  #endif
>
>  static constexpr int MIN_CORE_FDS = MIN_LEVELDB_FDS + NUM_FDS_MESSAGE_CAPTURE;
> -static const char* DEFAULT_ASMAP_FILENAME="ip_asn.map";
> +static const char* DEFAULT_ASMAP_FILENAME = "ip_asn.map";
>
> Thanks for the PR! I notice there are some whitespace changes that appear
> unrelated to the main goal of fixing documentation references.
> While these changes don't affect functionality, it would be helpful to
> keep the diff focused on the actual documentation URL changes. This makes
> the PR easier to review and reduces the risk of merge conflicts.
>
> Could you please remove these unrelated whitespace changes? This will make
> the PR's intent clearer and follow the principle of keeping changes minimal
> and focused.
>
> The core changes to fix the documentation references look good.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/bitcoin/bitcoin/pull/32699#pullrequestreview-2932549034>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AZVGWLD5HTPO47YKQ2QXSG33D3NEPAVCNFSM6AAAAAB6ZXKXIOVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDSMZSGU2DSMBTGQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

Zeegaths commented on src/init.cpp:
```diff
@@ -437,7 +438,18 @@ static void registerSignalHandler(int signal, void(*handler)(int))
+std::string GetDocumentationUrl(const std::string& doc_path)
+{
+    if (CLIENT_VERSION_MINOR == 99) {
+        std::string stable_version = strprintf("v%d.0", CLIENT_VERSION_MAJOR);
```
when i run this,  it goes to version 30, which does not exist yet..,

Zeegaths commented:
Yes, it's done.

On Fri, 18 Jul 2025 at 15:41, maflcko ***@***.***> wrote:

> *maflcko* left a comment (bitcoin/bitcoin#32699)
> <https://github.com/bitcoin/bitcoin/pull/32699#issuecomment-3089360188>
>
> Can this be drafted/closed, or are you still working on this?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/bitcoin/bitcoin/pull/32699#issuecomment-3089360188>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AZVGWLBQ4RMG7UWJVYM6WG33JDTOFAVCNFSM6AAAAAB6ZXKXIOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTAOBZGM3DAMJYHA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

Zeegaths commented:
thanks for pointing that out. all good now

On Mon, 21 Jul 2025 at 17:19, maflcko ***@***.***> wrote:

> *maflcko* left a comment (bitcoin/bitcoin#32699)
> <https://github.com/bitcoin/bitcoin/pull/32699#issuecomment-3096984862>
>
> No, you haven't replied/addressed the feedback from June: #32699 (comment)
> <https://github.com/bitcoin/bitcoin/pull/32699#discussion_r2150261914>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/bitcoin/bitcoin/pull/32699#issuecomment-3096984862>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AZVGWLHFAQADBZYLO2USSGD3JTZFVAVCNFSM6AAAAAB6ZXKXIOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTAOJWHE4DIOBWGI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

Zeegaths committed:
adds correct documentation links

adds correct documentation links

b-l-u-e commented on src/init.cpp:
```diff
@@ -438,6 +438,18 @@ static void registerSignalHandler(int signal, void(*handler)(int))
+    if (CLIENT_VERSION_MINOR == 99) {
+        std::string stable_version = strprintf("v%d.0", CLIENT_VERSION_MAJOR);
+        return strprintf("https://github.com/bitcoin/bitcoin/blob/%s/doc/%s", stable_version, doc_path);
+        ;
```
There's an unnecessary semicolon on its own line that should be removed.

Zeegaths commented:
Fixed!

On Tue, 22 Jul 2025 at 23:44, Winnie Gitau ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In src/init.cpp
> <https://github.com/bitcoin/bitcoin/pull/32699#discussion_r2223787842>:
>
> > @@ -438,6 +438,18 @@ static void registerSignalHandler(int signal, void(*handler)(int))
>  }
>  #endif
>
> +std::string GetDocumentationUrl(const std::string& doc_path)
> +{
> +    if (CLIENT_VERSION_MINOR == 99) {
> +        std::string stable_version = strprintf("v%d.0", CLIENT_VERSION_MAJOR);
> +        return strprintf("https://github.com/bitcoin/bitcoin/blob/%s/doc/%s", stable_version, doc_path);
> +        ;
>
> There's an unnecessary semicolon on its own line that should be removed.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/bitcoin/bitcoin/pull/32699#pullrequestreview-3044818666>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AZVGWLA4EKFVAEIR6Z77EPT3J2PCVAVCNFSM6AAAAAB6ZXKXIOVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZTANBUHAYTQNRWGY>
> .
> You are receiving this because you modified the open/close state.Message
> ID: ***@***.***>
>
